### PR TITLE
Reduce redundant code duplication in the solve methods

### LINKED ISSUE #123: Avoid redundant duplication of code segment for solution object in the solve methods
### Feature Description
**Please describe the feature you would like to see added:**

In the current situation, preparation of the  solution object are handled in the `_solve()` and `_solve_adam()` methods before returning it to `__call__()` method. 

It seems to be such repetitive code segments is redundant and must be eliminated by moving them to the `__call__()` method. 

### Rationale
**Why is this feature important? What problem does it solve?**

- Code maintenance 
- Code readability

### PR DESCRIPTION
**Pull request by MehmetC-1QBit**
Please see the [contributing guidelines](https://github.com/ccvm/master/CONTRIBUTING.md) for how to create and submit a high-quality PR for this repo.

### Description

> In `LangevinSolver`, `DLSolver`, and `MFSolver` classes, `solution` update is handled at the end of `_solve()` and `_solve_adam()` methods before returning the outcome to `__call__()` method. Such a redundancy must be avoided for an efficient code management including maintenance and readability. 

### References

Closes #123 

### Checklist

- [x]  All active GitHub checks for tests, formatting, and security are passing
- [x] The correct base branch is being used, if not main
- [x] Issue #123 is addressed in this PR
- [x] `Changelog` is update
- [x] Sample scripts in `ccvm/examples/` were executed for different options

### CODE REVIEW

File: ccvm_simulators/solvers/dl_solver.py [hunk 1]
```diff
@@ -257,7 +257,10 @@ def _solve(
                 Defaults to None, which generates a filename based on the problem instance name.
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:258]: Please also remember to remove `post_processor` argument from `_solve()` method and in its docstrings. This applies to all the solvers.

File: ccvm_simulators/solvers/dl_solver.py [hunk 2]
```diff
@@ -257,7 +257,10 @@ def _solve(
...
 
         Returns:
-            solution (Solution): The solution to the problem instance.
+            c, s (tensor): random variables 
+            c_sample, s_sample (tensor): random sample [needs attention]
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:261]: Could we be more specific on these docstrings? Why `[needs attention]` is there? Is this something work in progress?

File: ccvm_simulators/solvers/dl_solver.py [hunk 3]
```diff
@@ -299,6 +302,8 @@ def _solve(
         # Start the timer for the solve
         solve_time_start = time.time()
 
+        c_sample = None # Make it global
+        s_sample = None # Make it global
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:306]: Can we not make them global now?

File: ccvm_simulators/solvers/dl_solver.py [hunk 4]
```diff
@@ -482,7 +429,10 @@ def _solve_adam(
                 Defaults to None, which generates a filename based on the problem instance name.
 
         Returns:
-            solution (Solution): The solution to the problem instance.
+            c, s (tensor): random variables 
+            c_sample, s_sample (tensor): random sample [needs attention]
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:433]: Same comment as above.

File: ccvm_simulators/solvers/dl_solver.py [hunk 5]
```diff
@@ -523,7 +473,9 @@ def _solve_adam(
 
         # Start the timer for the solve
         solve_time_start = time.time()
-
+        
+        c_sample = None # Make it global
+        s_sample = None # Make it global
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:478]: Same comment as above.

File: ccvm_simulators/solvers/dl_solver.py [hunk 6]
```diff
@@ -482,7 +429,10 @@ def _solve_adam(
                 Defaults to None, which generates a filename based on the problem instance name.
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:429]: `post_processor` argument can be removed from the function as well as the docstirngs. Same for all the solvers.

File: ccvm_simulators/solvers/dl_solver.py [hunk 7]
```diff
@@ -839,3 +730,63 @@ def __call__(
             raise ValueError(
                 f"Solver option type {type(algorithm_parameters)} is not supported."
             )
+
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/dl_solver.py:733]: Nice work in reduce code redundancy! 👍

File: ccvm_simulators/solvers/langevin_solver.py [hunk 8]
```diff
@@ -235,7 +235,10 @@ def _solve(
                 Defaults to None, which generates a filename based on the problem instance name.
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/langevin_solver.py:236]: The comments on DL solvers apply to this solver as well.

File: ccvm_simulators/solvers/mf_solver.py [hunk 9]
```diff
@@ -291,7 +291,10 @@ def _solve(
                 the problem instance name.
 
         Returns:
-            solution (Solution): The solution to the problem instance.
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/solvers/mf_solver.py:294]: The comments on DL solvers apply to this solver as well.

File: ccvm_simulators/tests/test_mf_solver.py [hunk 10]
```diff
@@ -252,7 +252,8 @@ def test_solve_success_minimal_inputs(self):
         self.mf_solver.change_variables = self.mock_change_variables
         self.mf_solver.calculate_grads = self.mock_calculate_grads
 
-        solution = self.mf_solver._solve(instance)
+        # solution = self.mf_solver._solve(instance)
```
Comment (Estella-Wang-1qbit) [inline ccvm_simulators/tests/test_mf_solver.py:255]: Is this outdated code? Can we remove it?

File: examples/ccvm_boxqp_dl.py [hunk 11]
```diff
@@ -30,13 +30,13 @@
...
-            alpha=0.001, beta1=0.9, beta2=0.999, add_assign=False
+            alpha=0.001, beta1=0.9, beta2=0.999, # 1.0, # 
+            add_assign=False
         )
 
         solution = solver(
             instance=boxqp_instance,
-            post_processor=None,
-            algorithm_parameters=adam_parameters,  # Set to None to use the original DL algorithm
+            post_processor=None, # algorithm_parameters=adam_parameters,  # Set to None to use the original DL algorithm
```
Comment (Estella-Wang-1qbit) [inline examples/ccvm_boxqp_dl.py:39]: Let's split the commented code on the next line.

File: examples/ccvm_boxqp_mf.py [hunk 12]
```diff
@@ -37,13 +37,13 @@
...
-            alpha=0.001, beta1=0.9, beta2=0.999, add_assign=False
+            alpha=0.001, beta1=0.9, beta2=0.999, # 1.0, #   
+            add_assign=False
         )
 
         solution = solver(
             instance=boxqp_instance,
-            post_processor=None,
-            algorithm_parameters=adam_parameters,  # Set to None to use the original MF algorithm
+            post_processor=None, # algorithm_parameters=adam_parameters,  # Set to None to use the original MF algorithm
```
Comment (Estella-Wang-1qbit) [inline examples/ccvm_boxqp_mf.py:46]: Same here.

File: examples/langevin_boxqp.py [hunk 13]
```diff
@@ -35,12 +35,12 @@
...
         adam_parameters = AdamParameters(
-            alpha=0.001, beta1=0.9, beta2=0.999, add_assign=False
+            alpha=0.001, beta1=0.9, beta2=0.999, 
+            add_assign=False
         )
         solution = solver(
             instance=boxqp_instance,
-            post_processor=None,
-            algorithm_parameters=adam_parameters,  # Set to None to use the original Langevin algorithm
+            post_processor=None,  algorithm_parameters=adam_parameters,  # Set to None to use the original Langevin algorithm
```
Comment (Estella-Wang-1qbit) [inline examples/langevin_boxqp.py:43]: Same here.

**Review by Estella-Wang-1qbit — CHANGES_REQUESTED** [review]
Awesome work Mehmet. After the refactoring, the codebase is much cleaner. I have some cosmetic comments for you.

File: ccvm_simulators/solvers/dl_solver.py (hunk 1 above)
Comment (MehmetC-1QBit) [reply ccvm_simulators/solvers/dl_solver.py:255]: Thanks for letting me know this. Nov it is removed from the signature of both `_solve()` and `_solve_adam()`. The docstrings are updated accordingly.

File: ccvm_simulators/solvers/dl_solver.py (hunk 2 above)
Comment (MehmetC-1QBit) [reply ccvm_simulators/solvers/dl_solver.py]: During that time I wasn't sure how to fill out their description. Now it is fixed.

File: ccvm_simulators/solvers/dl_solver.py (hunk 3 above)
Comment (MehmetC-1QBit) [reply ccvm_simulators/solvers/dl_solver.py]: Comments causes the confusion. So I removed all of them.

File: examples/ccvm_boxqp_dl.py (hunk 11 above)
Comment (MehmetC-1QBit) [reply examples/ccvm_boxqp_dl.py]: Here is the screenshot of what I have done. Hopefully, the change will address your concern. If not let me know.
Screenshot 2023-11-20 at 5 26 05 PM

**Review by Estella-Wang-1qbit — APPROVED** [review]
Thank you Mehmet. This looks good to me! Great work!

**Pull request merged**